partcad_skyline/skyline.py

```
@@ -0,0 +1,146 @@
+    def clone_repo(self):
+        if os.path.exists(self.clone_dir):
+            logging.info(f"Directory {self.clone_dir} already exists. Skipping clone.")
+            return
+        try:
+            # logging.info('Repository being cloned to %s', self.clone_dir)
+            GitPython.Repo.clone_from(self.repo_url, self.clone_dir)
+            logging.info(f"Repository cloned to {self.clone_dir}")
+        except GitPython.GitCommandError as e:
+            logging.info(f"Error cloning repository: {e}")
+
+    def get_commit_counts_by_date(self, author: str = None):
```

Ruff (0.8.0)

Missing return type annotation for public function `get_commit_counts_by_date`

(ANN201)